src/core/utils/init_db.c

```diff
@@ -0,0 +1,98 @@
+#include "leveldb/c.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include "string.h"
+#include "init_db.h"
+
+int set_db_path(char **dest, char *name){
+  if(!getenv("HOME")){
+    exit(1);
+  }
+  char* home_path = getenv("HOME");
+  *dest = malloc(strlen(home_path)+strlen(name)+1);
+  strcpy(*dest, home_path);
+  strcat(*dest, name);
+  return 1;
+}
+
+int open_or_create_db(leveldb_t **db, char *path){
+  if(*db != NULL){
```

Comment:
This isn't a reliable check - uninitialized pointers aren't guaranteed to be NULL so this could evaluate to true even when we haven't initialized a DB. AFAIK, you actually can't check if a pointer has memory allocated to it, so I don't think we can totally avoid running into segfaults with the DB. Let's just let leveldb handle it here and be sure to always init our globals.